tests/test-user.js

```diff
@@ -571,6 +573,52 @@ describe('User', () => {
       });
     });
 
+    it('should log error and return undefined when GET OVH returns 500 error', (done) => {
+      utils.cleanMocks();
+      utils.mockUsers();
+      utils.mockSlack();
+      utils.mockOvhTime();
+
+      const getOvhEmailCreation = nock(/.*ovh.com/)
+        .get(/^.*email\/domain\/.*\/account\/.*/)
+        .reply(500);
+
+      knex('users').insert({
+        username: 'utilisateur.nouveau',
+        secondary_email: 'utilisateur.nouveau.perso@example.com',
+      }).then(async () => {
+        const shouldBeUndefined = await createEmailAddresses();
```

Comment:
Je me trompe peut être, mais il me semble que cet outcome est toujours `undefined`, même quand OVH répond favorablement, je ne voit pas l’intérêt de tester cela :thinking: 

Si ce que vous intéresse c'est de s'assurer que la fonction ne balance pas une exception non-gérée, il y a le `should.not.throw` dans la lib chai.

Reply:
Il renvoie un tableau avec les tasks d'email traitées si tout c'est bien passé, si je ne me trompe pas. 
`return await Promise.all(emailCreationTasks);
`
et `undefined` si on passe par le `catch`

Reply:
Tout à fait, tu as raison.

Juste une mini remarque du coup... en l'état actuel, si je change la ligne 584 pour que le `getOvhEmailCreation` retourne un 200 OK et non pas un 500, le test passe quand même, car l'appel d'après (le POST) balance une exception car il n'est pas mocké.

Du coup, j'ajouterai le mock pour le post aussi dans ce test - ce qui m'assurera que le test va bien échouer quand il devra. Vous pourriez même ajouter une petite ligne pour s'assurer que le nock POST n'a pas été appelé, genre : 

```javascript
     const getOvhEmailCreation = nock(/.*ovh.com/)
        .get(/^.*email\/domain\/.*\/account\/.*/)
        .reply(500);

      const postOvhEmailCreation = nock(/.*ovh.com/)
        .post(/^.*email\/domain\/.*\/account/)
        .reply(200);

      knex('users').insert({
        username: 'utilisateur.nouveau',
        secondary_email: 'utilisateur.nouveau.perso@example.com',
      }).then(async () => {
        const shouldBeUndefined = await createEmailAddresses();
        getOvhEmailCreation.isDone().should.be.true;
        postOvhEmailCreation.isDone().should.be.false;
        should.not.exist(shouldBeUndefined);
        done();
      });
```

Comme ça si le getOVH retourne un 200 le test échoue